packages/circuit-ui/components/Tabs/components/TabList/TabList.tsx

```diff
@@ -13,44 +13,161 @@
  * limitations under the License.
  */
 
-import { Children, forwardRef, type HTMLAttributes } from 'react';
+'use client';
+
+import {
+  Children,
+  forwardRef,
+  type HTMLAttributes,
+  type KeyboardEventHandler,
+  type MouseEventHandler,
+  useCallback,
+  useEffect,
+  useRef,
+} from 'react';
 
 import { clsx } from '../../../../styles/clsx.js';
 import { utilClasses } from '../../../../styles/utility.js';
+import { isArrowLeft, isArrowRight } from '../../../../util/key-codes.js';
 
 import classes from './TabList.module.css';
 
 export interface TabListProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * If true, the tabs will stretch to fill the width of their container.
+   * @default false
+   */
   stretched?: boolean;
 }
 
 const MOBILE_AUTOSTRETCH_ITEMS_MAX = 3;
 
+const getCurrentTab = (node?: HTMLElement | null) =>
+  node
+    ? node.querySelector<HTMLElement>('[role="tab"][aria-selected="true"]')
+    : undefined;
+
 /**
  * TabList component that wraps the Tab components
  */
 export const TabList = forwardRef<HTMLDivElement, TabListProps>(
-  ({ className, style = {}, stretched, children, ...props }, ref) => {
+  ({ className, stretched, children, onClick, onKeyDown, ...props }, ref) => {
+    const gliderRef = useRef<HTMLSpanElement>(null);
+    const tabListRef = useRef<HTMLDivElement>(null);
     const numberOfTabs = Children.toArray(children).length;
-    const tabWidth = Math.floor(100 / numberOfTabs);
     const stretchOnMobile = numberOfTabs <= MOBILE_AUTOSTRETCH_ITEMS_MAX;
+
+    const updateGliderStyles = useCallback((tab: HTMLElement) => {
+      const { offsetLeft, offsetWidth } = tab;
+      gliderRef.current?.style?.setProperty(
+        'transform',
+        `translateX(${offsetLeft}px)`,
+      );
+      gliderRef.current?.style?.setProperty('width', `${offsetWidth}px`);
+    }, []);
+
+    useEffect(() => {
+      // scrolls the active tab into view on initial render
+      const activeTab = getCurrentTab(tabListRef.current);
+      activeTab?.scrollIntoView({ behavior: 'smooth', inline: 'start' });
+    }, []);
```

Comment:
What if the tabs aren't in the initial viewport, e.g. because they're further down the page? This would scroll the page to the tabs, right? Or what if there are multiple tabs on the page? Which one takes precedence?

To avoid these issues, we could instead set the [`scrollLeft`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollLeft) property on the parent element.